src/components/Dropdown.js

```diff
@@ -0,0 +1,300 @@
+import CreatableSelect from 'react-select/creatable'
+import AsyncSelect from 'react-select/async'
+import AsyncCreatableSelect from 'react-select/async-creatable'
+import styled, { withTheme, css } from 'styled-components'
+import { components } from 'react-select'
+import customCursor from '../assets/custom-cursor.png'
+
+const dropdownWidth = {
+  normal: '300px',
+  searchable: '400px',
+}
+
+const sharedStyle = css`
+  .react-select__control {
+    background-color: transparent;
+    margin: 2em 0;
+    border: 2px solid;
+    border-radius: 7px;
+    box-shadow: none;
+    max-width: ${p => (p.isSearchable ? dropdownWidth.searchable : dropdownWidth.normal)};
+    height: 48px;
+    border-color: ${p => (p.isValid ? p.theme.colors.default : p.theme.colors.warning)};
+    padding-right: 17px;
+    padding-left: 7px;
+    flex-direction: ${p => p.isSearchable && `row-reverse`};
+    cursor: url(${customCursor}), auto;
+  }
+
+  .react-select__control:hover,
+  .react-select__control--is-focused {
+    border-color: ${p => (p.isValid ? p.theme.colors.primary : p.theme.colors.secondaryWarning)};
+  }
+
+  .react-select__option {
+    color: ${p => p.theme.colors.background};
+    overflow-wrap: break-word;
+  }
+
+  .react-select__menu {
+    background-color: ${p => p.theme.colors.primary};
+    border-radius: 7px;
+    margin-top: 0px;
+    max-width: ${p => (p.isSearchable ? dropdownWidth.searchable : dropdownWidth.normal)};
+    cursor: url(${customCursor}), auto;
+  }
+
+  .react-select__menu-list::-webkit-scrollbar {
+    width: 24px;
+    background: transparent;
+  }
+
+  .react-select__menu-list::-webkit-scrollbar-thumb {
+    border: solid 8px transparent;
+    background-clip: padding-box;
+    border-radius: 15px;
+    box-shadow: inset 0 0 14px 14px ${p => p.theme.colors.scrollbar};
+  }
+
+  .react-select__option {
+    cursor: url(${customCursor}), auto;
+  }
+
+  .react-select__option:hover,
+  .react-select__option--is-selected {
+    background-color: ${p => p.theme.colors.hover};
+    border-radius: 5px;
+    font-weight: bold;
+  }
+
+  .react-select__placeholder {
+    color: ${p => p.theme.colors.default};
+  }
+
+  .react-select__single-value,
+  .react-select__input {
+    color: ${p => p.theme.colors.primary};
+  }
+
+  .react-select__menu-notice--no-options {
+    padding: 0;
+    text-align: left;
+  }
+`
+
+const StyledCreatableDropdown = styled(CreatableSelect)`
+  ${sharedStyle}
+`
+
+const StyledNormalDropdown = styled(ReactSelect)`
+  ${sharedStyle}
+`
+
+const StyledAsyncDropdown = styled(AsyncSelect)`
+  ${sharedStyle}
+`
+
+const StyledAsyncCreatableDropdown = styled(AsyncCreatableSelect)`
+  ${sharedStyle}
+`
+
+const StyledUserMessage = styled.div`
+  color: ${p => p.theme.colors.background};
+  font-weight: bold;
+  padding: 8px 12px;
+`
+
+const StyledErrorMsg = styled.div`
+  color: ${p => p.theme.colors.warning};
+  font-weight: 400;
+  font-size: 16px;
+  padding: 8px 0px;
+  margin-top: -30px;
```

Comment:
```suggestion
  position: absolute;
  top: 56px;
```
I realized that the error message pushes the subsequent content downwards so the content underneath moves up when you click on the dropdown. I think you can make it fixed by having `postion: absolute`

#### before
![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/38872354/100148178-3ddc2b80-2e51-11eb-9669-d1f5eddf9b76.gif)

#### after
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/38872354/100147833-bdb5c600-2e50-11eb-8f33-6f77e18bda6f.gif)